ihatemoney/forms.py

```diff
@@ -44,6 +45,28 @@ def process_formdata(self, value):
         return super(CommaDecimalField, self).process_formdata(value)
 
 
+class CalculatorStringField(StringField):
+    """
+    A class to deal with math ops (+, -, *, /)
+    in StringField
+    """
+
+    def process_formdata(self, valuelist):
+        if valuelist:
+            error_msg = "Not a valid amount or expression"
+            value = str(valuelist[0]).replace(" ", "").replace(",", ".")
+
+            if not match(r'^[ 0-9\.\+\-\*/\(\)]{0,50}$', value) or "**" in value:
+                raise ValueError(error_msg)
+
+            try:
+                valuelist[0] = str(eval(value, {"__builtins__": None}, {}))
```

Comment:
I would rather avoid the `eval` and use a small dedicated parser instead. The current code with `eval` should be safe, but it is easy to miss it and make a mistake in a further edit of the file, contrary to a safer approach (in my opinion) with a parser.

Reply:
Thanks for the feature, this will be really useful :-)

After a quick research, it seems that using [`ast.literal_eval`](https://python.readthedocs.io/en/v2.7.2/library/ast.html?highlight=ast#ast.literal_eval) would avoid us exposing a broad scope to the users.

Reply:
Nice! though `ast.literal_eval` seems to only work for addition and subtraction.

Reply:
Ended up using [this for parsing](https://stackoverflow.com/a/9558001/5431797) =) (a5819e9)

Reply:
Nice!